src/cycles/eulerian.jl

```diff
@@ -0,0 +1,110 @@
+# Checking for existence of Eulerian circuit and Eulerian trail
+
+function isConnected(
+    graph::AbstractGraph{U},
+    ) where U<:Integer
+
+    visited = zeros(UInt8, nv(graph))
+
+    for v in vertices(graph)
+        if length(outneighbors(graph, v)) == 0
+            visited[v] = 2
+        end
+    end
+
+    for v in vertices(graph)
+
+        if length(outneighbors(graph, v)) == 0
+            continue
+        end
+        S = Vector{U}([v])
+        visited[v] = 1
+        while !isempty(S)
+            u = S[end]
+            w = 0
+            for n in outneighbors(graph, u)
+                if visited[n] == 0
+                    w = n
+                    break
+                end
+            end
+            if w != 0
+                visited[w] = 1
+                push!(S, w)
+            else
+                visited[u] = 2
+                pop!(S)
+            end
+        end
+
+        break
+    end
+
+    for v in vertices(graph)
+        if(visited[v]==0)
+            return false
+        end
+    end
+
+    return true
+end
+
+"""
+    eulerian_trail(g)
+
+Return `true` if graph `g` contains [Eulerian trail](https://en.wikipedia.org/wiki/Eulerian_path).
+
+### Implementation Notes
+Uses single call to iterative version of DFS.
+"""
+function eulerian_trail(
+    graph::AbstractGraph{U},
+    ) where U<:Integer
+
+    if(!isConnected(graph))
+        return false
+    end
+
+    odd = 0
+    for v in vertices(graph)
+        if (length(outneighbors(graph, v))%2 == 1)
```

Comment:
`length(outneighbors(g,v)))` is typically just `outdegree(g, v)`, and you're just checking `isodd()`.